src/user/user.service.spec.ts

```diff
@@ -0,0 +1,173 @@
+      expect(user.user.email).toBe('user@email.com');
+      expect(user.partnerAdmin).toBeUndefined();
+      expect(user.partnerAccesses).toBe(undefined);
+      expect(repoSpyCreate).toBeCalledWith(createUserDto);
+      expect(repoSpySave).toBeCalled();
+      expect(apiCall).toBeCalled();
+    });
+    it('when supplied with user dto and partner access, it should return a new partner user', async () => {
+      const repoSpyCreate = jest.spyOn(repo, 'create');
+      const repoSpySave = jest.spyOn(repo, 'save');
+      const now = new Date();
+      const partnerAccessSpy = jest
+        .spyOn(mockPartnerAccessService, 'assignPartnerAccessOnSignup')
+        .mockResolvedValue({
+          partner: { id: '123' } as PartnerEntity,
+          userId: '123',
+          partnerId: '123',
+          accessCode: '123456',
+          therapySession: [],
+          featureLiveChat: true,
+          featureTherapy: true,
+          therapySessionsRedeemed: 0,
+          therapySessionsRemaining: 6,
+          id: 'id',
+          active: true,
+          activatedAt: now,
+          updatedAt: now,
+          createdAt: now,
+        } as PartnerAccessEntity);
+      const partnerRepoSpy = jest
+        .spyOn(mockPartnerRepository, 'findOne')
+        .mockResolvedValue({ id: '123' } as PartnerEntity);
+
+      const user = await service.createUser({ ...createUserDto, partnerAccessCode: '123456' });
+      expect(user.user.email).toBe('user@email.com');
+      expect(user.partnerAdmin).toBeNull();
+      expect(user.partnerAccesses).toEqual([
+        {
+          accessCode: '123456',
+          activatedAt: now,
+          active: true,
+          createdAt: now,
+          featureLiveChat: true,
+          featureTherapy: true,
+          id: 'id',
+          partner: { id: '123' },
+          therapySessions: [],
+          therapySessionsRedeemed: 0,
+          therapySessionsRemaining: 6,
+          updatedAt: now,
+        },
+      ]); // TODO check this
+
+      expect(repoSpyCreate).toBeCalledWith(createUserDto);
+      expect(partnerAccessSpy).toBeCalled();
+      expect(partnerRepoSpy).toBeCalled();
+      expect(repoSpySave).toBeCalled();
+
+      expect(apiCall).toBeCalled();
+    });
+
+    it('when supplied with user dto and partner access that has already been used, it should return an error', async () => {
+      mockPartnerAccessService.assignPartnerAccessOnSignup.mockRejectedValue(
+        new HttpException(PartnerAccessCodeStatusEnum.ALREADY_IN_USE, HttpStatus.CONFLICT),
+      );
+      await expect(async () => {
+        await service.createUser({ ...createUserDto, partnerAccessCode: '123456' });
+      }).rejects.toThrow(PartnerAccessCodeStatusEnum.ALREADY_IN_USE);
+    });
+    // TODO - what do we want to happen here?
+    it('when supplied with user dto and partner access that has an incorrect partner, it should return a user without public access', async () => {
```

Comment:
Do you mean without partner access here? i.e. a public user

Reply:
It was just a bad data case. What happens if there is a partner access code which has a bad id for a partner (doubt this will ever happen)? I just started putting in niche error cases to see the behaviour

Reply:
I've left it as a question because we might want to think about it at some point but didn't want to solve it now

Reply:
Yes I did haha